eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -0,0 +1,117 @@
+    type: object
+    default: []
+  - name: BuildTargetingString
+    type: string
+    default: 'azure-*'
+  - name: AdditionalTestArgs
+    type: string
+    default: ''
+  - name: TestMarkArgument
+    type: string
+    default: ''
+  - name: InjectedPackages
+    type: string
+    default: ''
+  - name: BuildDocs
+    type: boolean
+    default: true
+  - name: JobName
+    type: string
+    default: 'Test'
+  - name: AllocateResourceGroup
+    type: boolean
+    default: true
+  - name: DeployArmTemplate
+    type: boolean
+    default: false
+  - name: TestTimeoutInMinutes
+    type: number
+    default: 120
+  - name: TestSamples
+    type: boolean
+    default: false
+  - name: Location
+    type: string
+    default: ''
+  - name: Matrix
+    type: object
+    default:
+      Linux_Python35:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: '3.5'
+        CoverageArg: '--disablecov'
+      MacOs_Python37:
+        Pool: Azure Pipelines
+        OSVmImage: 'macOS-10.15'
+        PythonVersion: '3.7'
+        CoverageArg: '--disablecov'
+      Windows_Python27:
+        Pool: azsdk-pool-mms-win-2019-general
+        OSVmImage: MMS2019
+        PythonVersion: '2.7'
+        CoverageArg: '--disablecov'
+      Linux_PyPy3:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: 'pypy3'
+        CoverageArg: '--disablecov'
+      Linux_Python39:
+        Pool: azsdk-pool-mms-ubuntu-1804-general
+        OSVmImage: MMSUbuntu18.04
+        PythonVersion: '3.9'
+        CoverageArg: ''
+  - name: CloudConfigurations
+    type: object
+    default:
+      AzureCloud:
+        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+
+
+stages:
+  - stage: Build
+    jobs:
+    - template: ../jobs/archetype-sdk-tests.yml
+      parameters:
+        ServiceDirectory: ${{ parameters.ServiceDirectory }}
+        EnvVars: ${{ parameters.EnvVars }}
+        MaxParallel: ${{ parameters.MaxParallel }}
+        BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
+        AfterTestSteps: ${{ parameters.AfterTestSteps }}
+        AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
+        BuildTargetingString: ${{ parameters.BuildTargetingString }}
+        TestMarkArgument: ${{ parameters.TestMarkArgument }}
+        InjectedPackages: ${{ parameters.InjectedPackages }}
+        BuildDocs: ${{ parameters.BuildDocs }}
+        JobName: ${{ parameters.JobName }}
+        AllocateResourceGroup: ${{ parameters.AllocateResourceGroup }}
+        DeployArmTemplate: ${{ parameters.DeployArmTemplate }}
+        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        TestSamples: ${{ parameters.TestSamples }}
+        Location: ${{ parameters.Location }}
+        Matrix: ${{ parameters.Matrix }}
+        CloudConfigurations: ${{ parameters.CloudConfigurations }}
+
+
+  - stage: Publish_Coverage
+    dependsOn: Build
```

Comment:
The dependsOn is unnecessary, since it will implicitly do the stages in list order. But as Wes asked, you can probably just have this in the job list instead.